File: codeiq/2513.py
# 256 exercise
def sumOfTwoInArray(arr, target_sum):
    for val in arr:
        if ((target_sum - val) in arr) and ((target_sum - val) != val):
            return "yes"
    return "no"

array_length = input()
vals = [int(i) for i in input().split()]

print(sumOfTwoInArray(vals, 256))

File: codeiq/2513.rb
# 256 ruby solution

def sumOfTwoInArray(arr, target_sum)
  arr.each { |val| 
    if (arr.include? (target_sum - val) and (val != (target_sum - val)))
      return "yes" 
    end
  }
  "no"
end

input_length = gets.chomp.to_i
input_array = gets.chomp.split(' ').map(&:to_i)

puts sumOfTwoInArray(input_array, 256)
